libvast/test/segment_store.cpp

```diff
@@ -52,11 +55,14 @@ struct fixture : fixtures::deterministic_actor_system_and_events {
 
   /// @returns all segment files of the segment stores.
   auto segment_files() {
-    std::vector<path> result;
-    vast::directory dir{segment_path};
-    for (auto file : dir)
-      if (file.is_regular_file())
-        result.emplace_back(std::move(file));
+    std::vector<std::filesystem::path> result;
+    std::error_code err{};
+    auto dir = std::filesystem::directory_iterator{segment_path, err};
+    if (!err) {
+      std::copy_if(std::filesystem::begin(dir), std::filesystem::end(dir),
+                   std::back_inserter(result),
+                   [&](const auto& entry) { return entry.is_regular_file(); });
+    }
```

Comment:
```suggestion
    auto dir = std::filesystem::directory_iterator{segment_path, err};
    REQUIRE(!err);
    std::copy_if(std::filesystem::begin(dir), std::filesystem::end(dir),
                 std::back_inserter(result),
                 [&](const auto& entry) { return entry.is_regular_file(); });
```

I haven't tested this, but failing on error might be what we want here. Otherwise we return less query results than expected.

Reply:
I just changed it to ensure `err` isn't true as you suggested. I'm fine with that approach. We don't have any tests that would get us in a case where `err` would be `true` from what I could tell, so it's not a biggie from my perspective.

Reply:
REQUIRE is basically an assertion, and I think asserting is absolutely correct here. The tests cannot continue if there is an error.

Reply:
I took a look at the code some more. This is in a utility function that gets called for tests on invalid stores as well, so not doing anything on error is absolutely correct. The returned vector will then be empty, which is checked for.